scripts/download_file.py

```diff
@@ -0,0 +1,245 @@
+    If a file 'remote_dir/remote_file.file_list' is present on the server, that means
+    that the file has been split and a list of the split files is returned.  If there
+    is no file ending in '.file_list' then an empty list is returned.  '.file_list'
+    is assumed to be the output of the 'md5sum' program used to generate hashes for
+    verification of file integrity.
+
+    Parameters
+    ----------
+        remote_dir
+            URL of directory on remote that contains files
+        remote_file
+            name of the file on the remote to download
+
+    Returns
+    -------
+        files
+            list of file names (not URL) on remote that are split
+        checksums
+            checksum (MD5) for each file on the remote
+    """
+
+    remote_file = remote_dir + "/" + remote_file + ".file_list"
+
+    file_contents = get_remote_file_contents(remote_file)
+    if not file_contents:
+        return [], {} 
+
+    lines = file_contents.splitlines()
+    checksums = {}
+    files = []
+    for line in lines:
+        parts = line.split(" *")
+        files.append(parts[1])
+        checksums[parts[1]] = parts[0]
+
+    return files, checksums
+
+
+def get_remote_checksum(remote_dir: str, remote_file: str) -> str:
+    """
+    Return the MD5 hash from the MD5 file corresponding to the remote file (not split).
+
+    Parameters:
+        remote_dir
+            URL of directory on remote that contains files
+        remote_file
+            name of the file on the remote to download
+
+    Returns
+    -------
+        MD5 hash of the remote file
+    """
+    remote_file = remote_dir + "/" + remote_file + ".md5"
+
+    file_contents = get_remote_file_contents(remote_file)
+    if not file_contents:
+        return ""
+
+    parts = file_contents.split(" ")
+    if parts[0]:
+        return parts[0]
+    else:
+        return ""
+
+
+def download_file(url: str, dest_file: str) -> bool:
+    """ 
+    Download a file from the remote and save it to the specified local file.
+
+    Parameters:
+        url
+            URL of the remote file
+        dest_file
+            local path to store downloaded file to
+    """
+
+    try:
+        u = urllib.request.urlopen(url)
+    except urllib.error.HTTPError as e:
+        print(f"Unable to urlopen {url}: {e}")
+        return False
+
+    with open(dest_file, "wb") as f:
+        while True:
+            buffer = u.read(BLOCK_SIZE)
+            if not buffer:
+                break
+            f.write(buffer)
+
+    return True
+
+
+def calculate_md5(file_path: str) -> str:
+    """
+    Calculate the MD5 hash for the file specified by the path.
+
+    Parameters
+    ----------
+        file_path
+            path to a file on the local file system
+
+    Returns
+    -------
+        MD5 hash of the file
+    """
+
+    hash_md5 = hashlib.md5()
+    with open(file_path, "rb") as fh:
+        while chunk := fh.read(BLOCK_SIZE):
```

Comment:
The `:=` operator is not supported in Python 3.6.  On an architectural level should we support older versions of Python 3?

Reply:
You kind of need to support older python within this script because it is used inside KBase. If that were not true, I would say that only supporting active versions of python (which [as of now is 3.9+](https://devguide.python.org/versions/#python-release-cycle)) would be the way to go. I think `:=` is a python3.8 feature but I might be wrong.

I think the code you previously had is fine given the version constraint.